internal/handler/executor.go

```diff
@@ -134,13 +138,60 @@ func (e *Executor) execHookCommand(w io.Writer) error {
 		// log only envs set by webhook, not global env; otherwise it's leaking secrets to logs
 		"environment", envs,
 		"working_directory", cmd.Dir,
+		"timeout", timeout,
 	)
 	cmd.Stderr = w
 	cmd.Stdout = w
-
+	// handling the timeout
+	if timeout > 0 {
+		// sets the same PGID for the child processes
+		setPGID(cmd)
+		terminationTimer := e.stopProcessWithTimeout(cmd, timeout)
+		defer func(t *time.Timer) {
+			// stop timer if command had finished before the timeout reached
+			t.Stop()
+			select {
+			case <-t.C:
+			default:
+			}
+		}(terminationTimer)
```

Comment:
my bad, seems like I overlooked this part of time.AfterFunc docs:

> The returned Timer's C field is not used and will be nil.

so it turns out that <-t.C will never yield as it is always nil for this object.
in that case it is enough to just call defer t.Stop() w/o any additional logic, which would've been necessary for raw time.Timer

```suggestion
        // stop the timer if process had terminated before timer reached
		defer terminationTimer.Stop()
```